service/round_test.go

```diff
@@ -209,10 +210,12 @@ func TestRound_State(t *testing.T) {
 
 	// Trigger cleanup.
 	r.broadcasted()
-	time.Sleep(1 * time.Second)
+	req.NoError(r.waitTeardown(context.Background()))
 
 	// Verify cleanup.
 	state, err = r.state()
 	req.EqualError(err, fmt.Sprintf("file is missing: %v", filepath.Join(r.datadir, roundStateFileBaseName)))
 	req.Nil(state)
+
+	stop()
```

Comment:
When creating `context.WithCancel` you can just call `defer stop()` to make sure it is called at least once (multiple calls don't cause problems). Otherwise this might not be called when a `require` check fails.

Reply:
I know, but in this test the context is recreated many times and `stop()` is called many times manually. I call the last one manually too for consistency.